rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -118,4 +127,40 @@ public static boolean isServerUp()
 
     return false;
   }
+
+  /**
+   * Starts a new thread to report the liveness metric at a fixed rate.
+   */
+  private static void scheduleLivenessMetric()
```

Comment:
I dont think you do not need these classes or the scheduled executor. You only need to register the gauge in the constructor. The statsdreporter (or any reporter) will call a get value regularly. You've already implemented the getValue function to return 1. Also the fact that you had to put the null condition in `registerLivenessCheck` is also a clue that you want to do something only once. 

So the change required is to move the gauge registration (Lines 155-162) to `startServer`

Reply:
Carry-over from initially implementing it as a counter. I agree with your notes, and will update the gauge resgistration.

Reply:
Gauge registration has been updated.